include/odp/api/spec/packet.h

```diff
@@ -1178,6 +1178,31 @@ int odp_packet_move_data(odp_packet_t pkt, uint32_t dst_offset,
  *
  */
 
+/**
+ * Flags to control packet data checksum checking
+ */
+typedef union odp_packet_parse_chksum_t {
+	/** Individual check bits. */
+	struct {
+		/** Check IPv4 header checksum */
+		uint32_t ipv4_chksum   : 1;
+
+		/** Check UDP checksum */
+		uint32_t udp_chksum    : 1;
+
+		/** Check TCP checksum */
+		uint32_t tcp_chksum    : 1;
+
+		/** Check SCTP checksum */
+		uint32_t sctp_chksum   : 1;
+
+	} check;
+
+	/** All check bits. This can be used to set/clear all flags. */
+	uint32_t all_check;
+
+} odp_packet_parse_chksum_t;
+
```

Comment:
Better name for this type would be odp_proto_chksums_t, just a bit field list of protocol checksums. Similar to other odp_proto_xxx_t types.

typedef union odp_packet_chksums_t {
	/** Individual checksum bits. */
	struct {
		/** IPv4 header checksum */
		uint32_t ipv4   : 1;

		/** UDP checksum */
		uint32_t udp    : 1;

		/** TCP checksum */
		uint32_t tcp    : 1;

		/** SCTP checksum */
		uint32_t sctp   : 1;

	} chksum;

	/** All checksum bits. This can be used to set/clear all flags. */
	uint32_t all_chksum;

} odp_packet_chksums_t;

Then odp_packet_parse_param_t would contain: proto, layer, check.

For example:
parse_param.check.chksum.ip = 1;
parse_param.check.chksum.udp = 1;

Reply:
@psavol updated

Reply:
There was actually multiple suggestions:
1) odp_proto_chksums_t == bit field of all checksums, chksum_t would be an enum of one checksum
2) struct == chksum (not check)
3) bits: ipv4, udp, tcp, sctp (not udp_chksum)
4) parser function API is new and still in api-next. So, parser_param_t could re-use this type. For example, parse_param.check.chksum.udp = 1

Reply:
@psavol sorry, it looks like I misviewed original comment and missed these suggestions. PR is updated now.

Number 4 is already in place, but we do not have API checks for packet checksum parsing (and it is not implemented anyway). I'm looking onto that part now.